utils/results_test.go

```diff
@@ -130,31 +135,66 @@ func TestGetSummary(t *testing.T) {
 				Scans: []formats.ScanSummaryResult{
 					{
 						Target: "target1",
-						ScaScanResults: &formats.ScaSummaryCount{
-							"Critical": formats.SummaryCount{"Undetermined": 1},
-							"High":     formats.SummaryCount{"Not Applicable": 1},
+						Vulnerabilities: &formats.ScanVulnerabilitiesSummary{
+							ScaScanResults:     &formats.ScanScaResult{SummaryCount: formats.TwoLevelSummaryCount{}},
+							SecretsScanResults: &formats.SummaryCount{"Low": 1},
+							SastScanResults:    &formats.SummaryCount{"Low": 1},
+						},
+						Violations: formats.TwoLevelSummaryCount{
+							formats.ViolationTypeSecurity.String(): formats.SummaryCount{"Critical": 1, "High": 1},
 						},
-						SecretsScanResults: &formats.SummaryCount{"Low": 1},
-						SastScanResults:    &formats.SummaryCount{"Low": 1},
 					},
 					{
 						Target: "target2",
-						ScaScanResults: &formats.ScaSummaryCount{
-							"Critical": formats.SummaryCount{"": 1},
-							"High":     formats.SummaryCount{"": 1},
+						Vulnerabilities: &formats.ScanVulnerabilitiesSummary{
+							ScaScanResults: &formats.ScanScaResult{
+								SummaryCount:   formats.TwoLevelSummaryCount{"Critical": formats.SummaryCount{"": 1}},
+								UniqueFindings: 1,
+							},
+							SecretsScanResults: &formats.SummaryCount{"Low": 1},
 						},
-						SecretsScanResults: &formats.SummaryCount{"Low": 1},
+						Violations: formats.TwoLevelSummaryCount{formats.ViolationTypeSecurity.String(): formats.SummaryCount{"High": 1}},
 					},
 				},
 			},
-			findingCount: 5,
+			findingCount: 6,
+			issueCount:   7,
 		},
 	}
 	for _, testCase := range testCases {
 		t.Run(testCase.name, func(t *testing.T) {
 			result := testCase.results.GetSummary()
 			assert.Equal(t, testCase.expected, result)
 			assert.Equal(t, testCase.findingCount, testCase.results.CountScanResultsFindings())
+			assert.Equal(t, testCase.issueCount, testCase.results.GetSummary().GetTotalIssueCount())
 		})
 	}
 }
+
+func getDummyScaTestResults(vulnerability, violation bool) (responses []services.ScanResponse) {
```

Comment:
Please add tests for violation which are not security for all of them (otherwise the use case of only violations and without vulnerabiliteis doesn't make sense)